### ISSUE TITLE
http3 client can't work after server restart

### ISSUE THREAD

**Issue by ldnvnbl** [lucas-clemente/quic-go#3627]
server:

```go
package main

import (
	"crypto/rand"
	"crypto/rsa"
	"crypto/tls"
	"crypto/x509"
	"encoding/pem"
	"fmt"
	"github.com/lucas-clemente/quic-go/http3"
	"math/big"
)

func GenerateTLSConfig() *tls.Config {
	key, err := rsa.GenerateKey(rand.Reader, 1024)
	if err != nil {
		panic(err)
	}
	template := x509.Certificate{SerialNumber: big.NewInt(1)}
	certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, &key.PublicKey, key)
	if err != nil {
		panic(err)
	}
	keyPEM := pem.EncodeToMemory(&pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(key)})
	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certDER})

	tlsCert, err := tls.X509KeyPair(certPEM, keyPEM)
	if err != nil {
		panic(err)
	}
	return &tls.Config{
		Certificates: []tls.Certificate{tlsCert},
	}
}

func main() {
	server := http3.Server{
		Addr:      "127.0.0.1:7979",
		TLSConfig: GenerateTLSConfig(),
	}
	err := server.ListenAndServe()
	fmt.Println("listen and serve err:", err)
}
```

client:

```go
package main

import (
	"crypto/tls"
	"fmt"
	"github.com/lucas-clemente/quic-go/http3"
	"io"
	"net/http"
	"time"
)

func main() {
	roundTripper := &http3.RoundTripper{
		TLSClientConfig: &tls.Config{
			InsecureSkipVerify: true,
		},
	}

	cli := http.Client{
		Transport: roundTripper,
	}

	cnt := 0
	for {
		cnt++
		resp, err := cli.Get("https://127.0.0.1:7979/ping")
		if err != nil {
			fmt.Println("cli get err:", err)
		}

		if resp != nil {
			data, err := io.ReadAll(resp.Body)
			if err != nil {
				fmt.Println("read all err:", err)
			}
			fmt.Println(string(data), cnt)
			resp.Body.Close()
		}
		time.Sleep(time.Second * 1)
	}
}
```

step1:  start server
step2:  start client: client should log "404 page not found" and increase number every second (this is expected)
step3: restart server:  client log nothing that means something block it, and wait some time, client log 'timeout: no recent network activity'

expect:

after step3, client should still work that do success http3 get, because server is running correctly just restart.

**Comment by marten-seemann** [comment]
Duplicate of #765.
